### ISSUE TITLE
Add support for HEIC photo

### ISSUE THREAD

**Issue by henri9813** [PrestaShop/PrestaShop#24494]
#### Is your feature request related to a problem?

I took my photo with an Iphone, then .... pictures are in HEIC format which is not currently accepted by Prestashop.

#### Describe the solution you'd like

When uploading a HEIC file, convert it immediatly to PNG file.

**Comment by matks** [comment]
@henri9813 I check HEIC format.

Imagemagick seems to be able to manipulate HEIC images but it is not used yet in the software. We have a pending issue to enable imagemagick https://github.com/PrestaShop/PrestaShop/issues/21101

To be added to https://github.com/PrestaShop/PrestaShop/issues/18518

**Comment by florine2623** [comment]
Hello @henri9813 ,

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thank you

**Comment by hibatallahAouadni** [comment]
Ping @matks is this improvement duplicate of #21101? or it should be fixed when #21101 will be fixed?

**Comment by matks** [comment]
> Ping @matks is this improvement duplicate of #21101? or it should be fixed when #21101 will be fixed?

When #21101 is fixed, we will be able to implement this issue 😉  you can see #21101 as a mandatory item to be done before starting this one

**Comment by hibatallahAouadni** [comment]
Thanks @matks :slightly_smiling_face:

**Comment by ldeprestashop** [comment]
Hello.

You can take directly JPEG pictures with your iPhone, so you will able to upload directly your picture on Prestashop.

https://www.techjunkie.com/heic-to-jpeg-iphone/

**Comment by henri9813** [comment]
Hello,

Do you really consider your option as an option for a  classical user ?

**Comment by ldeprestashop** [comment]
Hello.

Open the system IOS app and press a toggle (for chose JPEG instead HEIC)  seem more simple than use an eComerce BO : it's a simple solution to this problem.

Plus : this solution is available now ; you can use your iPhone now for direct use its photo.

**Comment by henri9813** [comment]
A simple user want a native solution.

He just want to upload photo :-) like myself or users wants.

But thanks for your purposal, i used to export my heic file to jpg with Photo application.

Thanks

**Comment by Hlavtox** [comment]
@matks @ldeprestashop Just got a call from customer who cannot upload a photo to Prestashop due to it being in HEIC... These formats are getting standard also on Androids, we need to implement it soon.

> Open the system IOS app and press a toggle (for chose JPEG instead HEIC) seem more simple than use an eComerce BO : it's a simple solution to this problem.

No comment

**Comment by andrea06590** [comment]
Hello, any news on this topic ?

**Comment by AlfredoPagano** [comment]
Hello, i think the HEIC support is really needed: i have a customer that wanna upload the photo from the iphone.
I modified the iphone for the photo-compatibility and now she can do it; but imagine if they use a different phone one day.... is not so much professional modify each phone they will use.... we should not do it, also for privacy.

The other platform allow it... why prestashop should be negatively different ? thats what a customer could say....

Let us know. Thanks
Alfredo.

**Comment by boo-code** [comment]
The dependency @matks named is still unmet, and the tracker makes it look otherwise - worth recording before anyone picks this up expecting a clear path.

#21101 "Add support for Imagemagick" reads as closed/completed, but its timeline shows it was `converted_to_discussion` and locked on 2023-04-21. It was moved, not implemented. The PR that would have done it, #28928 "Added support for ImageMagick" by @Progi1984, is closed unmerged, and there is no `Imagick` reference anywhere in the 9.2 tree - `grep -rn 'Imagick' src/ classes/` returns nothing.

So the prerequisite holds exactly as stated in 2021, and this issue is still behind it.

Current state of what the uploader accepts on 9.2, for reference:

```php
EXTENSIONS_SUPPORTED = ['gif', 'jpg', 'jpeg', 'jpe', 'png', 'webp'];
```

`avif` appears in `ImageManager` only as a destination format for resizing, not as something that can be uploaded. HEIC is not decodable by GD at all, so this is not a matter of adding a MIME type to that list - the file has to be converted by something, which is precisely what #21101 was about.

Two consequences worth stating:

The conversion-on-upload that this issue asks for is the smaller half. Once a decoder exists, converting to PNG or JPEG at upload time is a few lines in `ImageManager::resize()`'s neighbourhood; without one there is nothing to convert with.

And because #21101 now lives in Discussions, this issue currently has no open tracker item to block on. Either it gets relinked to that discussion, or the Imagick work is re-opened as an issue - otherwise this one reads as ready to start when it is not.

No patch prepared: there is nothing to build until the decoder question is settled, and settling it is the product decision that #21101 was moved out of the tracker without making.
